pkg/services/gcom/gcom.go

```diff
@@ -328,3 +336,43 @@ func (client *GcomClient) CreateToken(ctx context.Context, params CreateTokenPar
 
 	return token, nil
 }
+
+func (client *GcomClient) DeleteToken(ctx context.Context, params DeleteTokenParams) (bool, error) {
+	endpoint, err := url.JoinPath(client.cfg.ApiURL, "/v1/tokens", params.TokenID)
+	if err != nil {
+		return false, fmt.Errorf("building gcom tokens url: %w", err)
+	}
+
+	request, err := http.NewRequestWithContext(ctx, http.MethodDelete, endpoint, nil)
+	if err != nil {
+		return false, fmt.Errorf("creating http request: %w", err)
+	}
+
+	query := url.Values{}
+	query.Set("region", params.Region)
+
+	request.URL.RawQuery = query.Encode()
+	request.Header.Set("x-request-id", params.RequestID)
+	request.Header.Set("Authorization", fmt.Sprintf("Bearer %s", client.cfg.Token))
+
+	response, err := client.httpClient.Do(request)
+	if err != nil {
+		return false, fmt.Errorf("sending http request to delete access token: %w", err)
+	}
+	defer func() {
+		if err := response.Body.Close(); err != nil {
+			client.log.Error("closing http response body", "err", err.Error())
+		}
+	}()
+
+	if response.StatusCode == http.StatusNotFound {
+		return false, nil
```

Comment:
This seems a bit confusing to me. I get that we may not want to take the "token not found" as an error, but I feel is not clear that the problem is from the outside. I think I would prefer to remove the bool return value, and handle the "not happy" cases with errors, for instance, add an TokenNotFoundError and return that.

Reply:
Checking the CloudMigrationService, when we make this call, we ignore the bool return value, so there is no need to add this extra logic in my opinion. If we want to ignore the case where the token is not found, I would suggest to do that in the CloudMigrationService so is more clear (if that is the intention)

Reply:
Done. Returning an error instead of a boolean.